controllers/movie.controller.js

```diff
@@ -16,9 +16,52 @@ class MovieController {
         }
     }
 
-    async getAll() {
-        const movies = await this.movieRepository.find();
-        return movies.map(m => this.formatMovie(m));
+    async getPage(pagination, filter){
+        const movies = await this.getAll(pagination, filter);
+        const count = await this.getTotalCount(filter);
+
+        return {
+            movies: movies,
+            page: pagination.page,
+            totalCount: count
+        }
+    }
+
+    async getTotalCount(filter){
+        const query = {};
+        if (filter != undefined) {
+            if (filter.rating > 0 && filter.rating <= 5) {
+                query.rating = filter.rating;
+            }
+
+            if (filter.genres != null) {
```

Comment:
Rename the attribute `genres` to be just `genre` according to schema definition.